### PR TITLE
EntityData rewrite

### PR DESCRIPTION
**Pull request by OakLoaf**
I haven't changed the version number yet and I have also not tested this but `EntityData#create` is the new method to create the EntityData

- [x] Main changes
- [x] Backwards compatibility

### CODE REVIEW

**Comment by OakLoaf** [comment]
I still need to add backwards compat

File: src/main/java/me/xemor/configurationdata/comparison/RangeData.java [hunk 1]
```diff
@@ -18,7 +18,7 @@ public RangeData(String rangeStr) {
 
     public void init(String rangeStr) {
         if (rangeStr != null) {
-            String[] split = rangeStr.split("- ");
+            String[] split = rangeStr.replace(" ", "").split("-");
```
Comment (Xemorr) [inline src/main/java/me/xemor/configurationdata/comparison/RangeData.java]: negative numbers

**Comment by Xemorr** [comment]
@Septillihedron take a look at this

**Comment by OakLoaf** [comment]
Unsure what the final decision was regarding the "extra" section, are we going with moving it to the root or alternatively moving it into the "attributes" section as Sep suggested

**Comment by Xemorr** [comment]
> Unsure what the final decision was regarding the "extra" section, are we going with moving it to the root or alternatively moving it into the "attributes" section as Sep suggested

Attributes section is a terrible idea, attributes section is for things that are minecraft attributes. It's more misleading than extra.

File: src/main/java/me/xemor/configurationdata/entity/attribute/ZombifiableData.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import org.bukkit.entity.Hoglin;
+import org.bukkit.entity.PiglinAbstract;
+
+public class ZombifiableData extends EntityAttributeData {
+
+    private final boolean immuneToZombification;
+
+    public ZombifiableData(ConfigurationSection configurationSection) {
+        super(configurationSection);
+        if (configurationSection.contains("extra")) {
```
Comment (Xemorr) [inline src/main/java/me/xemor/configurationdata/entity/attribute/ZombifiableData.java]: DRY, move these checks to EntityData

**Comment by OakLoaf** [comment]
> > Unsure what the final decision was regarding the "extra" section, are we going with moving it to the root or alternatively moving it into the "attributes" section as Sep suggested
> 
> Attributes section is a terrible idea, attributes section is for things that are minecraft attributes. It's more misleading than extra.

I agree - it would get quite confusing and hard to manage

In that case I think this is all good for a proper in-game test

File: src/main/java/me/xemor/configurationdata/entity/attribute/ZombifiableData.java (hunk 2 above)
Comment (Xemorr) [reply src/main/java/me/xemor/configurationdata/entity/attribute/ZombifiableData.java]: also maybe a tweak to the message, just make the first bit be Deprecated: <path>then whatever else you say. I think it's useful to make it a little more succinct in console

Comment (OakLoaf) [reply src/main/java/me/xemor/configurationdata/entity/attribute/ZombifiableData.java]: If I do the checks in EntityData surely that wouldn't overwrite the variable in the subclasses?

**Review by Xemorr — APPROVED** [review]

**Pull request merged**